onair/src/ai_components/learners_interface.py

```diff
@@ -10,30 +10,22 @@
 """
 Data driven learning class for managing all data driven AI components
 """
-import importlib.util
-
+from ..util.plugin_import import import_plugins
 from ..util.data_conversion import *
 
 class LearnersInterface:
-    def __init__(self, headers, _ai_plugins={}):
+    def __init__(self, headers, _learner_plugins={}):
         assert(len(headers)>0), 'Headers are required'
         self.headers = headers
-        self.ai_constructs = []
-        for module_name in list(_ai_plugins.keys()):
-            spec = importlib.util.spec_from_file_location(module_name, _ai_plugins[module_name])
-            module = importlib.util.module_from_spec(spec)
-            spec.loader.exec_module(module)
-            self.ai_constructs.append(module.Plugin(module_name,headers))
-    
+        self.ai_constructs = import_plugins(self.headers, _learner_plugins)
+        
     def apriori_training(self, batch_data):
         for plugin in self.ai_constructs:
             plugin.apriori_training(batch_data)
 
-    def update(self, curr_data, status):
-        input_data = curr_data
-        output_data = status_to_oneHot(status)
+    def update(self, low_level_data, high_level_data):
         for plugin in self.ai_constructs:
-            plugin.update(input_data)
+            plugin.update(low_level_data)
```

Comment:
If we only are using low_level_data here, why are we putting high_level_data as an argument to the function?

Reply:
I think this was the kwargs issue we talked about, where the use of the abstracted class with default args wasnt working in the case of wanting to specific just high level data. So we over-standardized a bit

Reply:
We are actually passing the high level data at one point, so we'll keep it in for now and expect that we will enhance it later.